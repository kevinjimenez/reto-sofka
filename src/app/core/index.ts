export { spinnerInterceptor } from './interceptors/spinner.interceptor';
export { productsResolver } from './resolvers/products.resolver';
export { ProductsService } from './services/products.service';
export { SpinnerService } from './services/spinner.service';
