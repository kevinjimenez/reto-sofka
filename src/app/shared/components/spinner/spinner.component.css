.overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1000;
}

.loader {
  width: 48px;
  height: 48px;
  background: #fff;
  animation: rotate 1s linear infinite;
}

@keyframes rotate {
  0% {
    transform: rotate(0deg) scale(0.2);
    border-radius: 10%;
  }
  50% {
    transform: rotate(180deg) scale(1.5);
    border-radius: 50%;
  }
  100% {
    transform: rotate(360deg) scale(0.2);
    border-radius: 10%;
  }
}

