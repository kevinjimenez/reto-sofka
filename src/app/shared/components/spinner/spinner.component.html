@if (isLoading()) {
	<div class="overlay">
		<h1>Loading...</h1>
	</div>
}
