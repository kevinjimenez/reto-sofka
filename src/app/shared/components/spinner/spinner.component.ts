import { Component, inject } from '@angular/core';
import { SpinnerService } from '../../../core/services/spinner.service';

@Component({
	selector: 'app-spinner',
	standalone: true,
	imports: [],
	templateUrl: './spinner.component.html',
	styleUrl: './spinner.component.css'
})
export class SpinnerComponent {
	private readonly _spinnerService = inject(SpinnerService);

	public isLoading = this._spinnerService.isLoading;
}
