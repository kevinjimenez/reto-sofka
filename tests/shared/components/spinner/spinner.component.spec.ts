import { signal } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { SpinnerService } from '../../../../src/app/core/services/spinner.service';
import { SpinnerComponent } from '../../../../src/app/shared/components/spinner/spinner.component';

describe('SpinnerComponent', () => {
	let component: SpinnerComponent;
	let fixture: ComponentFixture<SpinnerComponent>;
	let compiled: HTMLElement;
	// let spinnerServiceMock: jest.Mocked<SpinnerService>;
	let spinnerServiceMock = {
		isLoading: signal<boolean>(false)
	};

	beforeEach(async () => {
		// spinnerServiceMock = {
		// 	isLoading: signal<boolean>(true)
		// } as jest.Mocked<SpinnerService>;

		await TestBed.configureTestingModule({
			imports: [SpinnerComponent],
			providers: [{ provide: SpinnerService, useValue: spinnerServiceMock }]
		}).compileComponents();
	});

	beforeEach(() => {
		fixture = TestBed.createComponent(SpinnerComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
		compiled = fixture.nativeElement;
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should match snapshot', () => {
		expect(compiled).toMatchSnapshot();
	});

	// it('should show the spinner when isLoading is true', () => {
	// 	const expected = 'Loading...';
	// 	component.isLoading.set(true);
	// 	fixture.detectChanges();

	// 	const contend = compiled.querySelector('.overlay');
	// 	expect(contend?.textContent).toBe(expected);
	// });
});
