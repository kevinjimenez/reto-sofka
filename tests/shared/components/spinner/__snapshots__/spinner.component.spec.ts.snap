// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerComponent should match snapshot 1`] = `
<div
  id="root1"
>
  
</div>
`;
